### ISSUE TITLE
bug: The regular expression does not fully match the output of the tool-call.

### ISSUE THREAD

**Issue by L-jasmine** [LlamaEdge/LlamaEdge#204]
### Summary

In regular expressions, the dot ‘.’ does not include newline characters, but there may be newline characters in the output of the LLM.
https://github.com/LlamaEdge/LlamaEdge/blob/9cb5e64f69926f01be28b831de679837f4c5df6d/api-server/llama-core/src/chat.rs#L1340

### Reproduction steps

```
fn main(){
    let s = r#"{
  "name": "writeCodeToPageTsx",
  "parameters": {
    "title": "Chat App",
    "description": "Real-time messaging platform.",
    "code": "<div class='bg-gray-200 p-4'>Your chat app content goes here.</div>"
  }
}"#;
    let re = regex::Regex::new(r"^\{.*\}$").unwrap();

    println!("{}",re.is_match(s)); // false
}
```

### Screenshots

![image](https://github.com/user-attachments/assets/fcee4f9f-f64b-4a9c-a377-944c70aff15c)

### Any logs you want to share for showing the specific issue

[2024-08-10 14:11:58.411] [wasi_logging_stdout] [info] llama_core: llama_core::chat in llama-core/src/chat.rs:1328: raw input: {
  "name": "writeCodeToPageTsx",
  "parameters": {
    "title": "Chat App",
    "description": "Real-time messaging platform.",
    "code": "<div class='bg-gray-200 p-4'>Your chat app content goes here.</div>"
  }
}
[2024-08-10 14:11:58.423] [wasi_logging_stdout] [info] llama_core: llama_core::chat in llama-core/src/chat.rs:1420: parsed result: ParseResult { raw: "{\n  \"name\": \"writeCodeToPageTsx\",\n  \"parameters\": {\n    \"title\": \"Chat App\",\n    \"description\": \"Real-time messaging platform.\",\n    \"code\": \"<div class='bg-gray-200 p-4'>Your chat app content goes here.</div>\"\n  }\n}", content: None, tool_calls: [] }

### Model Information

all tool model 

### Operating system information

ubuntu 22

### ARCH

amd64

### CPU Information

i5

### Memory Size

8G

### GPU Information

RTX 3060

### VRAM Size

8G

**Comment by apepkuss** [comment]
@L-jasmine Could you please provide the info about the model used in your case? Thanks!

**Comment by L-jasmine** [comment]
Llama-3-Groq-8B-Tool-Use
Llama3.1-8B

**Comment by apepkuss** [comment]
@L-jasmine Could you please provide the prompt strings used in your test? You can get it by searching the keyword `prompt:` in the log. The prompt string is necessary for reproducing the error. Thanks a lot!

**Comment by L-jasmine** [comment]
https://github.com/LlamaEdge/LlamaEdge/pull/207

**Comment by apepkuss** [comment]
Fixed by PR #207
